### ISSUE TITLE
Error executing recurring job with JobLambda 

### ISSUE THREAD

**Issue by atrujillofalcon** [jobrunr/jobrunr#714]
### JobRunr Version

6.1.0

### JDK Version

OpenJDK Runtime Environment Zulu17.34+19-CA (build 17.0.3+7-LTS)

### Your SQL / NoSQL database

SQLServer 14.0.2037 or PostgreSQL 13.9

### What happened?

After migrating from v5 to v6, my Spring microservice randomly started throwing this error:

```
java.lang.NullPointerException: Cannot call "Object.getClass()" because "<parameter1>" is null.
```

It is not deterministic, because if I restart the service, the error may go away.

I'm using jobrunr-spring-boot-2-starter (6.1.0) and Spring Boot 2.7.2 with JDK 17 and Maven 3.6.4

### How to reproduce?

```java
   Arrays.stream(FileType.values())
                .forEach(fileType -> jobScheduler.createRecurrently(RecurringJobBuilder.aRecurringJob()
                        .withId(fileType.getProcess())
                        .withCron(cron)
                        .withAmountOfRetries(5)
                        .withLabels("bde")
                        .withDetails((JobLambda) () -> getBdeFilesByFileType(fileType))));
```

### Relevant log output

```shell
2023-03-06 20:44:59.373  WARN [,,] 1 --- [pool-3-thread-2] org.jobrunr.server.JobZooKeeper          : JobRunr encountered a problematic exception. Please create a bug report (if possible, provide the code to reproduce this and the stacktrace) - Processing will continue.

java.lang.NullPointerException: Cannot invoke "Object.getClass()" because "<parameter1>" is null
	at java.base/java.util.Collections.unmodifiableSet(Unknown Source)
	at org.jobrunr.jobs.AbstractJob.getLabels(AbstractJob.java:83)
	at org.jobrunr.jobs.RecurringJob.toJob(RecurringJob.java:125)
	at org.jobrunr.jobs.RecurringJob.toScheduledJobs(RecurringJob.java:77)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.getJobsToSchedule(ProcessRecurringJobsTask.java:72)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.processRecurringJobs(ProcessRecurringJobsTask.java:51)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.runTask(ProcessRecurringJobsTask.java:33)
	at org.jobrunr.server.zookeeper.tasks.ZooKeeperTask.run(ZooKeeperTask.java:47)
	at org.jobrunr.server.JobZooKeeper.lambda$runMasterTasksIfCurrentServerIsMaster$0(JobZooKeeper.java:76)
	at java.base/java.util.Arrays$ArrayList.forEach(Unknown Source)
	at org.jobrunr.server.JobZooKeeper.runMasterTasksIfCurrentServerIsMaster(JobZooKeeper.java:76)
	at org.jobrunr.server.JobZooKeeper.run(JobZooKeeper.java:56)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.runAndReset(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

2023-03-06 20:45:14.378  WARN [,,] 1 --- [pool-3-thread-2] org.jobrunr.server.JobZooKeeper          : JobRunr encountered a problematic exception. Please create a bug report (if possible, provide the code to reproduce this and the stacktrace) - Processing will continue.

java.lang.NullPointerException: Cannot invoke "Object.getClass()" because "<parameter1>" is null
	at java.base/java.util.Collections.unmodifiableSet(Unknown Source)
	at org.jobrunr.jobs.AbstractJob.getLabels(AbstractJob.java:83)
	at org.jobrunr.jobs.RecurringJob.toJob(RecurringJob.java:125)
	at org.jobrunr.jobs.RecurringJob.toScheduledJobs(RecurringJob.java:77)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.getJobsToSchedule(ProcessRecurringJobsTask.java:72)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.processRecurringJobs(ProcessRecurringJobsTask.java:51)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.runTask(ProcessRecurringJobsTask.java:33)
	at org.jobrunr.server.zookeeper.tasks.ZooKeeperTask.run(ZooKeeperTask.java:47)
	at org.jobrunr.server.JobZooKeeper.lambda$runMasterTasksIfCurrentServerIsMaster$0(JobZooKeeper.java:76)
	at java.base/java.util.Arrays$ArrayList.forEach(Unknown Source)
	at org.jobrunr.server.JobZooKeeper.runMasterTasksIfCurrentServerIsMaster(JobZooKeeper.java:76)
	at org.jobrunr.server.JobZooKeeper.run(JobZooKeeper.java:56)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.runAndReset(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

2023-03-06 20:45:29.382  WARN [,,] 1 --- [pool-3-thread-2] org.jobrunr.server.JobZooKeeper          : JobRunr encountered a problematic exception. Please create a bug report (if possible, provide the code to reproduce this and the stacktrace) - Processing will continue.

java.lang.NullPointerException: Cannot invoke "Object.getClass()" because "<parameter1>" is null
	at java.base/java.util.Collections.unmodifiableSet(Unknown Source)
	at org.jobrunr.jobs.AbstractJob.getLabels(AbstractJob.java:83)
	at org.jobrunr.jobs.RecurringJob.toJob(RecurringJob.java:125)
	at org.jobrunr.jobs.RecurringJob.toScheduledJobs(RecurringJob.java:77)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.getJobsToSchedule(ProcessRecurringJobsTask.java:72)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.processRecurringJobs(ProcessRecurringJobsTask.java:51)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.runTask(ProcessRecurringJobsTask.java:33)
	at org.jobrunr.server.zookeeper.tasks.ZooKeeperTask.run(ZooKeeperTask.java:47)
	at org.jobrunr.server.JobZooKeeper.lambda$runMasterTasksIfCurrentServerIsMaster$0(JobZooKeeper.java:76)
	at java.base/java.util.Arrays$ArrayList.forEach(Unknown Source)
	at org.jobrunr.server.JobZooKeeper.runMasterTasksIfCurrentServerIsMaster(JobZooKeeper.java:76)
	at org.jobrunr.server.JobZooKeeper.run(JobZooKeeper.java:56)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.runAndReset(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

2023-03-06 20:45:44.386  WARN [,,] 1 --- [pool-3-thread-2] org.jobrunr.server.JobZooKeeper          : JobRunr encountered a problematic exception. Please create a bug report (if possible, provide the code to reproduce this and the stacktrace) - Processing will continue.

java.lang.NullPointerException: Cannot invoke "Object.getClass()" because "<parameter1>" is null
	at java.base/java.util.Collections.unmodifiableSet(Unknown Source)
	at org.jobrunr.jobs.AbstractJob.getLabels(AbstractJob.java:83)
	at org.jobrunr.jobs.RecurringJob.toJob(RecurringJob.java:125)
	at org.jobrunr.jobs.RecurringJob.toScheduledJobs(RecurringJob.java:77)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.getJobsToSchedule(ProcessRecurringJobsTask.java:72)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.processRecurringJobs(ProcessRecurringJobsTask.java:51)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.runTask(ProcessRecurringJobsTask.java:33)
	at org.jobrunr.server.zookeeper.tasks.ZooKeeperTask.run(ZooKeeperTask.java:47)
	at org.jobrunr.server.JobZooKeeper.lambda$runMasterTasksIfCurrentServerIsMaster$0(JobZooKeeper.java:76)
	at java.base/java.util.Arrays$ArrayList.forEach(Unknown Source)
	at org.jobrunr.server.JobZooKeeper.runMasterTasksIfCurrentServerIsMaster(JobZooKeeper.java:76)
	at org.jobrunr.server.JobZooKeeper.run(JobZooKeeper.java:56)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.runAndReset(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

2023-03-06 20:45:59.390  WARN [,,] 1 --- [pool-3-thread-2] org.jobrunr.server.JobZooKeeper          : JobRunr encountered a problematic exception. Please create a bug report (if possible, provide the code to reproduce this and the stacktrace) - Processing will continue.

java.lang.NullPointerException: Cannot invoke "Object.getClass()" because "<parameter1>" is null
	at java.base/java.util.Collections.unmodifiableSet(Unknown Source)
	at org.jobrunr.jobs.AbstractJob.getLabels(AbstractJob.java:83)
	at org.jobrunr.jobs.RecurringJob.toJob(RecurringJob.java:125)
	at org.jobrunr.jobs.RecurringJob.toScheduledJobs(RecurringJob.java:77)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.getJobsToSchedule(ProcessRecurringJobsTask.java:72)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.processRecurringJobs(ProcessRecurringJobsTask.java:51)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.runTask(ProcessRecurringJobsTask.java:33)
	at org.jobrunr.server.zookeeper.tasks.ZooKeeperTask.run(ZooKeeperTask.java:47)
	at org.jobrunr.server.JobZooKeeper.lambda$runMasterTasksIfCurrentServerIsMaster$0(JobZooKeeper.java:76)
	at java.base/java.util.Arrays$ArrayList.forEach(Unknown Source)
	at org.jobrunr.server.JobZooKeeper.runMasterTasksIfCurrentServerIsMaster(JobZooKeeper.java:76)
	at org.jobrunr.server.JobZooKeeper.run(JobZooKeeper.java:56)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.runAndReset(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

2023-03-06 20:46:14.397 ERROR [,,] 1 --- [pool-3-thread-2] org.jobrunr.server.JobZooKeeper          : FATAL - JobRunr encountered too many processing exceptions. Shutting down.

org.jobrunr.JobRunrException: JobRunr encountered a problematic exception. Please create a bug report (if possible, provide the code to reproduce this and the stacktrace)
	at org.jobrunr.JobRunrException.shouldNotHappenException(JobRunrException.java:43)
	at org.jobrunr.server.JobZooKeeper.run(JobZooKeeper.java:62)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.runAndReset(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.NullPointerException: Cannot invoke "Object.getClass()" because "<parameter1>" is null
	at java.base/java.util.Collections.unmodifiableSet(Unknown Source)
	at org.jobrunr.jobs.AbstractJob.getLabels(AbstractJob.java:83)
	at org.jobrunr.jobs.RecurringJob.toJob(RecurringJob.java:125)
	at org.jobrunr.jobs.RecurringJob.toScheduledJobs(RecurringJob.java:77)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.getJobsToSchedule(ProcessRecurringJobsTask.java:72)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.processRecurringJobs(ProcessRecurringJobsTask.java:51)
	at org.jobrunr.server.zookeeper.tasks.ProcessRecurringJobsTask.runTask(ProcessRecurringJobsTask.java:33)
	at org.jobrunr.server.zookeeper.tasks.ZooKeeperTask.run(ZooKeeperTask.java:47)
	at org.jobrunr.server.JobZooKeeper.lambda$runMasterTasksIfCurrentServerIsMaster$0(JobZooKeeper.java:76)
	at java.base/java.util.Arrays$ArrayList.forEach(Unknown Source)
	at org.jobrunr.server.JobZooKeeper.runMasterTasksIfCurrentServerIsMaster(JobZooKeeper.java:76)
	at org.jobrunr.server.JobZooKeeper.run(JobZooKeeper.java:56)
	... 6 common frames omitted

2023-03-06 20:46:14.398  INFO [,,] 1 --- [pool-3-thread-2] org.jobrunr.server.BackgroundJobServer   : BackgroundJobServer and BackgroundJobPerformers - stopping (waiting for all jobs to complete - max 10 seconds)
2023-03-06 20:46:24.415  INFO [,,] 1 --- [pool-3-thread-2] org.jobrunr.server.BackgroundJobServer   : JobRunr BackgroundJobServer shutdown requested - waiting for jobs to finish (at most 10 seconds)
2023-03-06 20:46:24.416  INFO [,,] 1 --- [pool-3-thread-2] org.jobrunr.server.BackgroundJobServer   : BackgroundJobServer and BackgroundJobPerformers stopped
```

**Comment by rdehuyss** [comment]
Which Json mapper are you using? I assume Jackson but I cannot reproduce this currently...

**Comment by rdehuyss** [comment]
Found it - it's not related to your new RecurringJobs mentioned above but related to some old recurring jobs that are still in your DB.

**Comment by atrujillofalcon** [comment]
It makes sense, they could be jobs from the old version of JobRunr v5 before the application was migrated to version 6. How should I clean up these corrupted jobs?

They might be calling methods that no longer exist in the class in which they were executed.

**Comment by atrujillofalcon** [comment]
Before to use the new fluent API I was using the old good @Recurring annotation with Jobrunr v5

**Comment by atrujillofalcon** [comment]
> 

I needed to create jobs dynamically depending on a list of values. Instead of having N annotations, I switched to the builder to have a single dynamic configuration.

**Comment by rdehuyss** [comment]
Fixed with commit 41af1a2.

**Comment by rdehuyss** [comment]
I checked this again and there is normally an internal JobRunr migration that migrates all your old Recurring jobs to the new format. For some reason, that must have failed for you.
